backend-rust/src/indexer.rs

```diff
@@ -672,59 +691,86 @@ impl PreparedBlock {
         })
     }
 
-    async fn save(
-        &self,
+    async fn batch_save(
+        batch: &Vec<Self>,
         context: &BlockProcessingContext,
```

Comment:
Maybe just take ownership of the context since you just clone it below. Let the caller clone if necessary.

Reply:
In case of something failing we need to rollback the context as well, hence the need for cloning it.
But I'll move it up and make it mutable here instead, to avoid returning anything.